ecosystem/nft-metadata-crawler/nft-metadata-crawler-parser/src/parser.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright © Aptos Foundation
+
+use crate::models::nft_metadata_crawler_uris::NFTMetadataCrawlerURIs;
+use diesel::{
+    r2d2::{ConnectionManager, PooledConnection},
+    PgConnection,
+};
+use nft_metadata_crawler_utils::NFTMetadataCrawlerEntry;
+
+/**
+ * Stuct that represents a parser for a single entry from queue
+ */
+#[allow(dead_code)]
+pub struct Parser {
+    entry: NFTMetadataCrawlerEntry,
+    model: NFTMetadataCrawlerURIs,
+    bucket: String,
+    token: String,
+    conn: PooledConnection<ConnectionManager<PgConnection>>,
+    cdn_prefix: String,
+}
```

Comment:
Wait why? Each entry needs a connection pool and all these metadata?

Reply:
It's just a single connection for postgres. I have a pool in the main function and providing it to each thread. I think that's ok? 

Also not sure how to better pass the metadata over? Would it be ok to just directly pass the parsed yaml file over because I'm just using the metadata from that?